astrology.lic

```diff
@@ -24,31 +24,40 @@ class Astrology
     @divination_bones_container = settings.divination_bones_storage['container']
     @divination_bones_tied = settings.divination_bones_storage['tied']
     @have_telescope = settings.have_telescope
-    @ap_training = settings.astral_plane_training['train_in_ap']
     @ap_source = settings.astral_plane_training['train_source']
     @ap_destination = settings.astral_plane_training['train_destination']
-    @predict_event = settings.predict_event
-    @full_pools = settings.astrology_use_full_pools
+    @prediction_pool_target = settings.astrology_use_full_pools ? 10 : 1 
+    @rtr_data = nil
+    echo " Flags['rtr-expire'].nil? #{Flags['rtr-expire'].nil?}" if UserVars.astrology_debug
+    echo " Flags['rtr-expire'] = #{Flags['rtr-expire']}" if UserVars.astrology_debug
+
     do_buffs(settings)
     if args.rtr
-      check_ripples
+      check_ripples(settings)
     else
-      train_astrology
+      train_astrology(settings)
     end
   end
 
   def do_buffs(settings)
+    return unless settings
+
+    # Pop out rtr data from buffs and save it for later
     if settings.waggle_sets['astrology']
-      wait_for_script_to_complete('buff', ['astrology'])
-    else
-      # this should probably be deprecated, this doesn't allow you to NOT cast the spells if they are up already
-      settings.astrology_buffs['spells'].each do |spell|
-        if spell['abbrev'].casecmp('rtr').zero?
-          next if !Flags['rtr-expire'].nil? && !Flags['rtr-expire']
-        end
-        echo "Buffing: #{spell}" if UserVars.astrology_debug
-        buff(spell, settings)
-      end
+      buffs = settings.waggle_sets['astrology']
+      @rtr_data = buffs.select { |spell| spell.eql?('Read the Ripples') }
+                       .values
+                       .first
+      buffs.reject! { |spell| spell.eql?('Read the Ripples') }
+      cast_spells(buffs, settings)
+    elsif settings.astrology_buffs
+      echo "******************"
+      echo "astrology_buffs has been deprecated in favor of 'astrology' waggle set and will be removed in a future version"
+      echo "******************"
+      buffs = settings.astrology_buffs['spells']
+      @rtr_data = buffs.select { |spell| spell['abbrev'].casecmp('rtr').zero? }.first
+      buffs.reject! { |spell| spell['abbrev'].casecmp('rtr').zero? }
```

Comment:
I would have to do reassignment then right? I assumed `!` let me do it in-place.